app/jobs/benchmark_regression_job.rb

```diff
@@ -0,0 +1,29 @@
+class BenchmarkRegressionJob < ActiveJob::Base
+  queue_as :default
+
+  def perform(initiator, benchmark_type, benchmark_result_type, category)
+
+    current_commit = BenchmarkRun.find_by(initiator: initiator, initiator_type: Commit, benchmark_type: benchmark_type, benchmark_result_type: benchmark_result_type)       
+
+    current_initiator = current_commit.initiator_id
+
+    last_initiator = BenchmarkRun.where(initiator_type: Commit, benchmark_type: benchmark_type,
+                     benchmark_result_type: benchmark_result_type).order(:created_at).first.initiator_id
+
+    benchmark_runs = BenchmarkRun.where(initiator_type: Commit, benchmark_type: benchmark_type,
+                     benchmark_result_type: benchmark_result_type).offset(last_initiator - current_initiator).limit(1000)
```

Comment:
This sets the offset so that the benchmarks which came after our benchmark are not considered.